web/impact/impact/v1/events/organization_created_event.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime
+from impact.models import (
+    Organization,
+    Partner,
+    Startup,
+)
+from impact.utils import DAWN_OF_TIME
+
+
+class OrganizationCreatedEvent(object):
+    EVENT_TYPE = "created"
+
+    def __init__(self, organization):
+        self.organization = organization
+
+    @classmethod
+    def events(cls, organization):
+        return [cls(organization)]
+
+    def serialize(self):
+        earliest, latest = self._created_datetime()
+        return {
+            "datetime": earliest,
+            "latest_datetime": latest,
+            "event_type": self.EVENT_TYPE,
+            "description": "",
+            }
+
+    def _created_datetime(self):
+        startup = self.organization.startup_set.first()
+        if startup:
+            return _created_datetimes(startup, Startup)
+        partner = self.organization.partner_set.first()
+        if partner:
+            return _created_datetimes(partner, Partner)
+        return _created_datetimes(self.organization, Organization)
+
+
+def _created_datetimes(instance, cls):
+    if instance.created_at is not None:
+        return (instance.created_at, instance.created_at)
+    if hasattr(instance, "created_datetime"):
+        if instance.created_datetime is not None:
+            return (instance.created_datetime,
+                    instance.created_datetime)
+        return (_previous_created_datetime(instance, cls),
+                _next_created_datetime(instance, cls))
+    next_instance = cls.objects.filter(id__gt=instance.id,
+                                       created_at__isnull=False
+                                       ).order_by("id").first()
+    if next_instance:
+        return (DAWN_OF_TIME, next_instance.created_at)
```

Comment:
This assumes that if a given instance has `created_at=None` than necessarily all the instances before that will also have a `created_at=None`? I don't like this assumption since there's no enforcement that it is indeed so (for example, we could've just as well populated the created_at backwards according to some heuristic).

I think it's worth the minor effort to implement this in a way that consults the DB explicitly.

Reply:
This is not a real case in the past or going forward, but I added it.